numpyro/infer/autoguide.py

```diff
@@ -561,26 +574,25 @@ def __call__(self, *args, **kwargs):
             if site["type"] != "sample" or site["is_observed"]:
                 continue
 
-            event_dim = self._event_dims[name]
             init_loc = self._init_locs[name]
+            event_dim = self._event_dims[name]
             with ExitStack() as stack:
                 for frame in site["cond_indep_stack"]:
                     stack.enter_context(plates[frame.name])
 
                 site_loc = numpyro.param(
-                    "{}_{}_loc".format(name, self.prefix),
-                    init_loc,
-                    constraint=site["fn"].support,
-                    event_dim=event_dim,
+                    f"{name}_{self.prefix}_loc", init_loc, event_dim=event_dim
                 )
-
-                site_fn = dist.Delta(site_loc).to_event(event_dim)
```

Comment:
Here you are finding the MAP point in unconstrained space. This class gets MAP point in constrained space.